ethicml/data/vision_data/cmnist.py

```diff
@@ -0,0 +1,129 @@
+    final_mask = np.zeros_like(targets, dtype=np.bool_)
+    for index, label in enumerate(classes_to_keep):
+        mask = targets == label
+        targets = np.where(mask, index, targets)
+        final_mask |= mask
+    dataset.targets = targets
+    inds = final_mask.nonzero()[0].tolist()
+
+    return Subset(dataset, inds)
+
+
+def create_cmnist_datasets(
+    *,
+    root: str,
+    scale: float,
+    test_pcnt: float,
+    download: bool = False,
+    seed: int = 42,
+    rotate_data: bool = False,
+    shift_data: bool = False,
+    padding: bool = False,
+    quant_level: int = 8,
+    input_noise: bool = False,
+    classes_to_keep: Optional[Sequence[_Labels]] = None,
+) -> Tuple[LdTransformedDataset, LdTransformedDataset]:
+    """Create and return colourised MNIST train/test pair.
+
+    Args:
+        root: Where the images are downloaded to.
+        scale: The amount of 'bias' in the colour. Lower is more biased.
+        test_pcnt: The percentage of data to make the test set.
+        download: Whether or not to download the data.
+        seed: Random seed for reproducing results.
+        rotate_data: Whether or not to rotate the training images.
+        shift_data: Whether or not to shift the training images.
+        padding: Whether or not to pad the training images.
+        quant_level: the number of bins to quantize the data into.
+        input_noise: Whether or not to add noise to the training images.
+        classes_to_keep: Which digit classes to keep. If None or empty then all classes will be kept.
+    Returns: tuple of train and test data as a Dataset.
+
+    """
+    np.random.seed(seed)
+    random.seed(seed)
+    torch.manual_seed(seed)
+
+    base_aug = [transforms.ToTensor()]
+    data_aug = []
+    if rotate_data:
+        data_aug.append(transforms.RandomAffine(degrees=15))
+    if shift_data:
+        data_aug.append(transforms.RandomAffine(degrees=0, translate=(0.11, 0.11)))
+    if padding > 0:
+        base_aug.insert(0, transforms.Pad(padding))
+    if quant_level != 8:
+        base_aug.append(Quantize(int(quant_level)))
+    if input_noise:
+        base_aug.append(NoisyDequantize(int(quant_level)))
+
+    mnist_train = MNIST(root=root, train=True, download=download)
+    mnist_test = MNIST(root=root, train=False, download=download)
+
+    if classes_to_keep:
+        mnist_train = _filter_classes(dataset=mnist_train, classes_to_keep=classes_to_keep)
+        mnist_test = _filter_classes(dataset=mnist_test, classes_to_keep=classes_to_keep)
+
+    all_data: ConcatDataset = ConcatDataset([mnist_train, mnist_test])
+
+    dataset_size = len(all_data)
+    indices = list(range(dataset_size))
+    split = int(np.floor((1 - test_pcnt) * dataset_size))
+
+    np.random.shuffle(np.array(indices))
+
+    train_indices, test_indices = indices[:split], indices[split:]
+
+    train_data = Subset(all_data, indices=train_indices)
+    test_data = Subset(all_data, indices=test_indices)
```

Comment:
split with `torch.utils.data.random_split`?

Reply:
Why not.